storage/spanstore/interface.go

```diff
@@ -51,4 +51,5 @@ type TraceQueryParameters struct {
 	DurationMin   time.Duration
 	DurationMax   time.Duration
 	NumTraces     int
+	NumSpans      int
```

Comment:
Hi @yurishkuro, I think this particular issue was filed for OOM on the jaeger-query nodes, and we have a separate issue for handling overloading of the underlying storage - #960

Reply:
OK. Will look into. Closing this PR.

So do we want to restrict number of spans (per trace) at ingestion?

Reply:
Not at ingestion (which isn't possible anyway since spans arrive async). We want the limit to be applied at query time, similar to the changes you made to `plugin/storage/es/spanstore/reader.go`, but the value of the limit to be configurable as a parameter of the query-service, passed at Reader construction from CLI, not as a parameter of the HTTP request.

Reply:
Sorry its taking so many iterations to get this one right. Understood now, sending in a new, clean PR.